dev/com.ibm.ws.channelfw/src/com/ibm/websphere/channelfw/osgi/CHFWBundle.java

```diff
@@ -514,19 +528,37 @@ protected synchronized void setBehavior(ServiceReference<HttpProtocolBehavior> r
 
         protocolBehaviorRef = reference;
         httpVersionSetting = (String) reference.getProperty(HttpProtocolBehavior.HTTP_VERSION_SETTING);
+        if (OPTIONAL_DEFAULT_OFF_20.equalsIgnoreCase(httpVersionSetting)) {
+            default20Off = true;
+            versionSet = true;
+        } else if (OPTIONAL_DEFAULT_ON_20.equalsIgnoreCase(httpVersionSetting)) {
+            default20On = true;
+            versionSet = true;
+        }
     }
 
     protected synchronized void unsetBehavior(ServiceReference<HttpProtocolBehavior> reference) {
         if (reference == this.protocolBehaviorRef) {
             protocolBehaviorRef = null;
             httpVersionSetting = null;
+            versionSet = false;
+            default20Off = false;
+            default20On = false;
         }
     }
 
     public static String getServletConfiguredHttpVersionSetting() {
         return httpVersionSetting;
     }
 
+    public static boolean isDefaultOff20() {
```

Comment:
@mrsaldana Do you have any thoughts on my changes to CHFWBundle in light of its websphere package?

Reply:
Any thoughts on the method name change to be clearer what the methods do?

Reply:
I believe components that import the CHF bundle com.ibm.ws.channelfw would be the only ones with access to this package. I think its meant to be mostly consumed by the transport (SSL/HTTP channels). While it was added for H2, I believe that the getServletConfiguredHttpVersionSetting was purposefully left general, in case  the channels would have to do specific behavior based on future Servlet versions. That being said, the two APIs being proposed here would be specific to HTTP/2, so I'd agree that at that point 

isHttp2DisabledByDefault and isHttp2EnabledByDefault would be easier (more clear) to read than isDefaultOn20 and isDefaultOff20.